src/mappings/nfts/setMetadata.ts

```diff
@@ -51,11 +51,7 @@ export async function handleMetadataSet(context: Context): Promise<void> {
         warn(OPERATION, `collection ${event.collectionId} not found`)
         return
       }
-      const nft = final as NFTEntity
-      const token = await handleTokenEntity(context, collection, nft)
-      if (token) {
-        nft.token = token
-      }
+      (final as NFTEntity).token =  await handleTokenEntity(context, collection, final as NFTEntity);
```

Comment:
Why need to retype?

Reply:
Because token is only on nftEntity
But final may be nftEntity or collectionEntity

This block is executed inside the scope of an if that checks that it is an nftEntity, but ts is not convinced, hence the retype